### PR TITLE
shellbags fixes & usrclass support

### PR DESCRIPTION
**Pull request by lotansery7**
- Fixed path concatenation bug
- Add support of Root Folder shell type
- Add known GUIDs list to improve path parsing
- Add support of Usrclass hive ShellBags

### CODE REVIEW

File: regipy/plugins/ntuser/shellbags_ntuser.py [hunk 1]
```diff
@@ -3,18 +3,406 @@
 from regipy.hive_types import NTUSER_HIVE_TYPE
 from regipy.plugins.plugin import Plugin
 from regipy.utils import convert_wintime
+from plaso.helpers.windows import shell_folders
```
Comment (mkorman90) [inline regipy/plugins/ntuser/shellbags_ntuser.py:6]: This is a really large dependency.  Is there any way to do this without it?

File: regipy/plugins/ntuser/shellbags_ntuser.py [hunk 2]
```diff
@@ -3,18 +3,406 @@
...
 import re
 
 logger = logging.getLogger(__name__)
 
 NTUSER_SHELLBAG = '\\Software\\Microsoft\\Windows\\Shell\\BagMRU'
-USRCLASS_SHELLBAG = '\\Local Settings\\Software\\Microsoft\\Windows\\Shell\\BagMRU'
 CODEPAGE = 'cp1252'
 
+known_guids = {
```
Comment (mkorman90) [inline regipy/plugins/ntuser/shellbags_ntuser.py:14]: Please move this to a const file besides the plugin.

**Review by mkorman90 — CHANGES_REQUESTED** [review]

File: regipy/plugins/ntuser/shellbags_ntuser.py (hunk 2 above)
Comment (lotansery7) [reply regipy/plugins/ntuser/shellbags_ntuser.py:14]: There is specific file to move it to? or to create new one? (and in which path?)

File: regipy/plugins/ntuser/shellbags_ntuser.py (hunk 1 above)
Comment (lotansery7) [reply regipy/plugins/ntuser/shellbags_ntuser.py]: Done

File: regipy/plugins/ntuser/shellbags_ntuser.py (hunk 2 above)
Comment (lotansery7) [reply regipy/plugins/ntuser/shellbags_ntuser.py]: Done

**Review by mkorman90 — APPROVED** [review]

**Pull request merged**